aten/src/ATen/NestedTensorImpl.cpp

```diff
@@ -180,26 +182,32 @@ NestedTensorImpl::NestedTensorImpl(
       "in the near future.");
   auto storage_device = storage_.device();
   TORCH_INTERNAL_ASSERT(
-      storage_device.is_cpu() || storage_device.is_cuda(),
-      "NestedTensorImpl storage must be either CUDA or CPU but got ",
+      storage_device.is_cpu() || storage_device.is_cuda() || storage_device.is_meta(),
+      "NestedTensorImpl storage must be CUDA, CPU, or Meta; got ",
       storage_device);
-  validate_nested_tensor_metadata(nested_sizes_, nested_strides_, storage_offsets_);
-  refresh_dim();
+  if (validate_metadata) {
+    validate_nested_tensor_metadata(nested_sizes_, nested_strides_, storage_offsets_);
```

Comment:
Wouldn't it make more sense to just noop this function when the tensors here are meta?

Reply:
Sure, that works. I was trying not to bake a lot of meta-specific logic in, but it does seem cleaner that way